plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -21,33 +45,288 @@ public static GlobalKafkaConfiguration get() {
+                        Jenkins.get(),
+                        StandardUsernamePasswordCredentials.class,
+                        Collections.singletonList(KAFKA_SCHEME),
+                        CredentialsMatchers.always()
+                )
+                .includeCurrentValue(credentialsId);
+    }
+
+    public FormValidation doCheckSslKeystoreCredentialsId(@AncestorInPath Item item, @QueryParameter String value) {
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return FormValidation.ok();
+            }
+        }
+        if (value.startsWith("${") && value.endsWith("}")) {
+            return FormValidation.warning("Cannot validate expression based credentials");
+        }
+        if (CredentialsProvider.listCredentials(
+                StandardUsernamePasswordCredentials.class,
+                Jenkins.get(),
+                ACL.SYSTEM,
+                Collections.singletonList(KAFKA_SCHEME),
+                CredentialsMatchers.withId(value)
+        ).isEmpty()) {
+            return FormValidation.error("Cannot find currently selected credentials");
+        }
+        this.sslKeystoreCredentialsId = value;
+        return FormValidation.ok();
+    }
+
+    public ListBoxModel doFillSslKeyCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(credentialsId);
+            }
+        }
+        return result
+                .includeMatchingAs(
+                        ACL.SYSTEM,
+                        Jenkins.get(),
+                        StandardUsernamePasswordCredentials.class,
+                        Collections.singletonList(KAFKA_SCHEME),
+                        CredentialsMatchers.always()
+                )
+                .includeCurrentValue(credentialsId);
+    }
+
+    public FormValidation doCheckSslKeyCredentialsId(@AncestorInPath Item item, @QueryParameter String value) {
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return FormValidation.ok();
+            }
+        }
+        if (value.startsWith("${") && value.endsWith("}")) {
+            return FormValidation.warning("Cannot validate expression based credentials");
+        }
+        if (CredentialsProvider.listCredentials(
+                StandardUsernamePasswordCredentials.class,
+                Jenkins.get(),
+                ACL.SYSTEM,
+                Collections.singletonList(KAFKA_SCHEME),
+                CredentialsMatchers.withId(value)
+        ).isEmpty()) {
+            return FormValidation.error("Cannot find currently selected credentials");
+        }
+        this.sslKeyCredentialsId = value;
+        return FormValidation.ok();
```

Comment:
There are many copy-paste cases in the code above. Utility methods could simplify it